identity/src/main/java/com/android/identity/util/ApplicationData.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.identity.util;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import co.nstant.in.cbor.model.Number;
+
+/**
+ * An interface to an object storing application data in a key-value pair manner.
+ *
+ * <p>This interface exists to support applications which wish to store additional data it wants to
+ * associate with a credential, for example the document type (e.g. DocType for 18013-5 credentials),
+ * user-visible name, logos/background, state, and so on.
+ */
```

Comment:
This makes it sound like it's only useful on credentials. I think we want to say `s/a credential/an object/` and `s/for example/for example on a {@link Credential} object one could use this to store/`.

Reply:
Good point